### PR TITLE
Blog: api authentication techniques

### PR DESCRIPTION
**Pull request by Shulammite-Aso**

### CODE REVIEW

**Comment by Shulammite-Aso** [comment]
@subomi

File: src/app/(main)/blog/articles/api-authentication-techniques.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+Who is using it: Many API providers use API keys, including big players like [Google Maps API](https://developers.google.com/maps/documentation/javascript/get-api-key) and [Stripe](https://docs.stripe.com/api/authentication). What's common among successful API providers using API keys is that they implement other measures to help their consumers keep their API keys secure. For instance, the Google Maps API platform makes provision for consumers to restrict the usage of their API keys to only certain applications, or IP addresses. This way a stolen API key cannot be used unless the request is coming from within the bounds of the restriction. 
+
+Stripe equally implements a similar feature that allows one to restrict the IP addresses that can make API calls using an API key. In addition, they implement key Rolling. Rolling a key revokes it and generates a replacement key.  Keys can be rolled manually or via a schedule that rolls the key after a certain time.
+
+![A diagram illustrating API key restriction by IP address](downloads/restricted-api-keys.png)
```
Comment (Oluwadaminiola) [inline src/app/(main)/blog/articles/api-authentication-techniques.md:53]: @Shulammite-Aso the images should be in the `blog-assets` plus they don't seem to be a part of the pr

Comment (Shulammite-Aso) [reply src/app/(main)/blog/articles/api-authentication-techniques.md]: Done this

**Review by Oluwadaminiola — APPROVED** [review]

**Review by subomi — CHANGES_REQUESTED** [review]
**Feedback**

1. We should close out the article with a CTA, you can check this [post](https://getconvoy.io/blog/interview-with-stanislav-zmiev) for an example.
2. When you generate the banner images, include them at the top of the post as well, similar to the post referenced in number 1.

**Review by subomi — APPROVED** [review]

**Pull request merged**